src/main/java/io/github/opencubicchunks/cubicchunks/core/world/cube/Cube.java

```diff
@@ -335,25 +335,18 @@ public void tickCubeServer(BooleanSupplier tryToTickFaster, Random rand) {
 
         tickCubeCommon(tryToTickFaster);
 
-        Iterator<NextTickListEntry> pti = pendingTickListEntriesTreeSet.iterator();
-        while (pti.hasNext()) {
+        while (!pendingTickListEntriesTreeSet.isEmpty()) {
             assert storage != null : "There are blocks to tick in cube but getStorage() is empty!";
-            NextTickListEntry ntle = pti.next();
+            NextTickListEntry ntle = pendingTickListEntriesTreeSet.first();
             if (ntle.scheduledTime > world.getTotalWorldTime())
                 return;
```

Comment:
This is why it didn't work before. If this is reached, no entries will be moved from hashset to treeset.

Reply:
Ow. Should I just replace `return` with `break` then? Where was a reason why I add this `256` ticks limit.